content/03.ml-concepts.md

```diff
@@ -1,24 +1,28 @@
 ## Tip 1: Concepts that apply to machine learning also apply to deep learning {#concepts}
-Deep learning is a distinct subfield of machine learning, but it is still a subfield.
-DL has proven to be an extremely powerful paradigm capable of outperforming “traditional” machine learning approaches in certain contexts, but it is not immune to the many limitations inherent to machine learning.
-Many best practices for machine learning also apply to deep learning.
-Like all computational methods, deep learning should be applied in a systematic manner that is reproducible and rigorously tested.
 
-Those developing deep learning models should select data that are relevant to the problem at hand; non-salient data can hamper performance or lead to spurious conclusions.
-Biases in testing data can also unduly influence measures of model performance, and it may be difficult to directly identify confounders from the model.
-Investigators should consider the extent to which the outcome of interest is likely to be predictable from the input data and begin by throughly inspecting the input data.
-Suppose that there are robust heritability estimates for a phenotype that suggest that the genetic contribution is modest but a deep learning model predicts the phenotype with very high accuracy.
+While deep learning has inarguably surpassed general machine learning for artificial intelligence and predictive modeling applications in terms of public awareness, deep learning still remains a subfield of machine learning.
+DL has proven to be an extremely powerful paradigm capable of outperforming "traditional" or "conventional" machine learning approaches. This is partly owed to the fact that DL algorithms are less dependent on manual feature extraction and are capable of achieving high predictive performance when operating on raw-data, such as images and text. directly.
+In contrast, traditional machine learning approaches often require manual feature extraction to produce acceptable results.
+However, the majority of best practices for machine learning still apply to deep learning.
+Like all computational methods, experiments involving deep learning should be conducted in a systematic manner and accompanied by reproducible instructions as well as rigorous model evaluation.
+
+The ability of deep learning to perform implicit feature extraction can be regarded as a double-edged sword.
+For instance, those developing deep learning models not only need to ensure that the data contains information relevant to the problem at hand but is also free of systematic biases.
+As deep learning models are capable of extracting and learning from complex patterns in a training dataset, DL models are also susceptible to systematic noise and non-salient information, which can not only hamper performance but also lead to spurious conclusions.
```

Comment:
I would argue that many ML methods are capable of extracting near-arbitrary patterns from the data, but DL introduces additional complexity because it can scale so well. With DL you end up with data of such high dimensionality that it is nearly impossible to examine the impact of each component of the input. Furthermore, the size of datasets used for DL are often quite large, and a thorough examination of their biases (which is generally domain-specific) may take much longer than selecting and developing the model.

Reply:
> I would argue that many ML methods are capable of extracting near-arbitrary patterns from the data, but DL introduces additional complexity because it can scale so well. 

Hm, but I'd say the majority of methods, like Logistic regression, linear SVMs, decision trees,  etc. can't, though. But yeah, but kernel SVMs with non-polynomial kernels do e.g., some sort of implicit feature extraction ...

> DL introduces additional complexity because it can scale so well. 

Hm, do you mean scaling with regard to a large numbers of features? In my opinion, that's only because of the implicit feature extraction. E.g., in convnets you have the convolutional kernels that combine features (vs. doing it manually) and you also have weight sharing to actually AVOID operating on a large number of features for making predictions. Usually, the (final) prediction layer is only ~100-2000 features large; I see everything that comes before that as implicit feature extraction layers.

You can actually empirically show that by e.g., implementing an autoencoder and then training a simple logistic regression classifier on the latent space. It will usually perform reasonably well on this compressed feature space (thanks to the feature extraction) whereas it would perform much worse if you'd trained it on the high-dimensional input layer directly.

Reply:
My guess is you won't be applying linear SVMs to the same problems that you will use CNNs for (: I agree that feature extraction is probably why NNs can succeed on some of these datasets, but I think the high dimensionality of the data at hand may be the real "core" of the problem. I guess I am just hesitant to make non-DL methods sound "safe", when they may be just as uninterpretable when applied to the same high dimensional data :/

Reply:
Oh I see, I think I misunderstood your concern. I was thinking about it like this

- traditional: image dataset -> extract landmarks (e.g., facial landmarks) -> apply RF/SVM
- DL: image dataset -> apply convnet (let it take care of feature extraction)

If you apply a regular MLP to large-feature sets, you would have the same problem as you'd have when applying e.g. SVM -- but then, I think just using a multilayer perceptron on tabular data wouldn't be really deep learning (I don't think though that someone would apply these methods to raw data though).

Reply:
I think the concern should be covered in the sentence:

> In contrast, traditional machine learning approaches often require manual feature extraction to produce acceptable results.